tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterDescriptionTests.cs

```diff
@@ -69,15 +90,24 @@ public void CreateInitial_should_return_initial_description()
         public void Constructor_should_initialize_instance()
         {
             var dnsMonitorException = new Exception();
-            var subject = new ClusterDescription(
+#pragma warning disable CS0618
+            var connectionModeSwitch = ConnectionModeSwitch.NotSet;
+#pragma warning restore CS0618
+            var subject = CreateClusterDescription(
                 __clusterId,
+#pragma warning disable CS0618
                 ClusterConnectionMode.ReplicaSet,
+                connectionModeSwitch, 
+#pragma warning restore CS0618
+                null,
                 dnsMonitorException,
                 ClusterType.ReplicaSet,
                 new[] { __serverDescription1, __serverDescription2 });
 
             subject.ClusterId.Should().Be(__clusterId);
+#pragma warning disable CS0618
             subject.ConnectionMode.Should().Be(ClusterConnectionMode.ReplicaSet);
+#pragma warning restore CS0618
```

Comment:
Parameterize constructor test method:

    #pragma warning disable 618
            [Theory]
            [InlineData(ConnectionModeSwitch.NotSet, ClusterConnectionMode.Automatic, null)]
            [InlineData(ConnectionModeSwitch.UseConnectionMode, ClusterConnectionMode.Automatic, null)]
            [InlineData(ConnectionModeSwitch.UseConnectionMode, ClusterConnectionMode.ReplicaSet, null)]
            [InlineData(ConnectionModeSwitch.UseDirectConnection, ClusterConnectionMode.Automatic, null)]
            [InlineData(ConnectionModeSwitch.UseDirectConnection, ClusterConnectionMode.Automatic, false)]
            [InlineData(ConnectionModeSwitch.UseDirectConnection, ClusterConnectionMode.Automatic, true)]
            public void Constructor_should_initialize_instance(ConnectionModeSwitch connectionModeSwitch, ClusterConnectionMode connectionMode, bool? directConnection)
            {
                var dnsMonitorException = new Exception();
                var subject = new ClusterDescription(
                    __clusterId,
                    connectionMode,
                    connectionModeSwitch,
                    directConnection,
                    dnsMonitorException,
                    ClusterType.ReplicaSet,
                    new[] { __serverDescription1, __serverDescription2 });

                subject.ClusterId.Should().Be(__clusterId);
                subject.ConnectionModeSwitch.Should().Be(connectionModeSwitch);
                switch (connectionModeSwitch)
                {
                    case ConnectionModeSwitch.UseConnectionMode: subject.ConnectionMode.Should().Be(connectionMode); break;
                    case ConnectionModeSwitch.UseDirectConnection: subject.DirectConnection.Should().Be(directConnection); break;
                }
                subject.DnsMonitorException.Should().BeSameAs(dnsMonitorException);
                subject.LogicalSessionTimeout.Should().NotHaveValue();
                subject.Servers.Should().ContainInOrder(new[] { __serverDescription1, __serverDescription2 });
                subject.State.Should().Be(ClusterState.Disconnected);
                subject.Type.Should().Be(ClusterType.ReplicaSet);
            }
    #pragma warning restore 618